### PR TITLE
python3Packages.portalocker: 1.7.0 -> 2.3.0

### PR DESCRIPTION
**Pull request by Artturin**
###### Motivation for this change
Supersedes #117444
@jonringer @dotlambda 

###### Things done
had to add substituteInPlace's because upstream doesn't update them
`az --version` worked fine but i haven't dont use azure so i didn't test more

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>chia</li>
  </ul>
</details>
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>azure-cli</li>
    <li>python38Packages.applicationinsights</li>
    <li>python38Packages.azure-identity</li>
    <li>python38Packages.concurrent-log-handler</li>
    <li>python38Packages.msal-extensions</li>
    <li>python38Packages.portalocker</li>
    <li>python39Packages.applicationinsights</li>
    <li>python39Packages.azure-identity</li>
    <li>python39Packages.concurrent-log-handler</li>
    <li>python39Packages.msal-extensions</li>
    <li>python39Packages.portalocker</li>
  </ul>
</details>
chia was already failing https://hydra.nixos.org/build/146518756 #128498

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/azure-cli/python-packages.nix [hunk 1]
```diff
@@ -6,7 +6,7 @@ let
     prePatch = (attrs.prePatch or "") + ''
       rm -f azure_bdist_wheel.py tox.ini
       substituteInPlace setup.py \
-        --replace "cryptography>=2.3.1,<3.0.0" "cryptography"
+        --replace "cryptography>=3.2,<3.4" "cryptography"
```
Comment (Artturin) [inline pkgs/tools/admin/azure-cli/python-packages.nix:9]: updated this substituteInPlace https://hydra.nixos.org/build/146088369/nixlog/99
`substituteStream(): WARNING: pattern 'cryptography>=2.3.1,<3.0.0' doesn't match anything in file 'setup.py'`

File: pkgs/development/python-modules/portalocker/default.nix [hunk 2]
```diff
@@ -1,34 +1,31 @@
...
+  propagatedBuildInputs = [
+    redis
   ];
 
   checkInputs = [
     sphinx
-    pytest
+    pytestCheckHook
     pytestcov
     pytest-flake8
```
Comment (dotlambda) [inline pkgs/development/python-modules/portalocker/default.nix:27]:
```suggestion
```

File: pkgs/development/python-modules/portalocker/default.nix [hunk 3]
```diff
@@ -1,34 +1,31 @@
...
-      url = "https://github.com/WoLpH/portalocker/commit/42e4c0a16bbc987c7e33b5cbc7676a63a164ceb5.patch";
-      sha256 = "01mlr41nhh7mh3qhqy5fhp3br4nps745iy4ns9fjcnm5xhabg5rr";
-      excludes = [ "pytest.ini" ];
-    })
+  propagatedBuildInputs = [
+    redis
   ];
 
   checkInputs = [
     sphinx
```
Comment (dotlambda) [inline pkgs/development/python-modules/portalocker/default.nix:24]:
```suggestion
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
This should be three separate commits.

File: pkgs/development/python-modules/msal-extensions/default.nix [hunk 4]
```diff
@@ -25,6 +25,13 @@ buildPythonPackage rec {
     pathlib2
   ];
 
+  # upstream doesn't update this requirement probably because they use pip
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace "portalocker~=1.0" "portalocker" \
+      --replace "portalocker~=1.6" "portalocker"
```
Comment (dotlambda) [inline pkgs/development/python-modules/msal-extensions/default.nix:32]: We don't care about Windows.

**Review by dotlambda — COMMENTED** [review]
It would be nice to open upstream issues/PRs about the compatibility with portalocker 2.3.

Can you also update msal-extensions' homepage to https://github.com/AzureAD/microsoft-authentication-extensions-for-python?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128562` at 825d255e run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/20f50ee79ee422db1299b47298a5f8d2">chia</a></li>
  </ul>
</details>
<details>
  <summary>11 packages built successfully:</summary>
  <ul>
    <li>azure-cli</li>
    <li>python38Packages.applicationinsights</li>
    <li>python38Packages.azure-identity</li>
    <li>python38Packages.concurrent-log-handler</li>
    <li>python38Packages.msal-extensions</li>
    <li>python38Packages.portalocker</li>
    <li>python39Packages.applicationinsights</li>
    <li>python39Packages.azure-identity</li>
    <li>python39Packages.concurrent-log-handler</li>
    <li>python39Packages.msal-extensions</li>
    <li>python39Packages.portalocker</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `pytest-flake8`.
Near pkgs/development/python-modules/portalocker/default.nix:4:3:
```
  |
4 | , pytest-flake8
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `pytestcov`.
Near pkgs/development/python-modules/portalocker/default.nix:3:3:
```
  |
3 | , pytestcov
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/msal-extensions/default.nix:39:0:
```
   |
39 |     description = "The Microsoft Authentication Library Extensions (MSAL-Extensions) for Python";
   | ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Review by jonringer — APPROVED** [review]
LGTM

failures are broken on target branch
```
https://github.com/NixOS/nixpkgs/pull/128562

1 package failed to build:
chia

11 packages built:
azure-cli python38Packages.applicationinsights python38Packages.azure-identity python38Packages.concurrent-log-handler python38Packages.msal-extensions python38Packages.portalocker python39Packages.applicationinsights python39Packages.azure-identity python39Packages.concurrent-log-handler python39Packages.msal-extensions python39Packages.portalocker
```

**Pull request merged**